mffpy/writer.py

```diff
@@ -34,25 +34,31 @@ def __init__(self, filename: str):
         self.filename = filename
         self.files: Dict[str, Any] = {}
         self._bin_file_added = False
+        self.mffdir, self.ext = splitext(self.filename)
```

Comment:
Am I correct in guessing that the intent here is to populate / prepare the filename variable at the time the `Writer` is instantiated rather than when a call is made to actually write the file?

Reply:
Yes, because in order to stream a file to a directory that directory needs to be created first, so I needed some means of accessing the name of the directory to pass to the bin writer.